### ISSUE TITLE
Forbid `map` where list or set comprehension can be used

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#894]
# Rule request

## Thesis

These two code samples are identical:

```python
operands = list(map(itemgetter(0), tokens))
unique_operands = set(map(attrgetter('unique_version'), tokens))
```

And:

```python
operands = [token[0] for token in tokens]
unique_operands = {token.unique_version for token in tokens}
```

But, the second version is more python-friendly and way faster. We should enforce it.
This is a refactoring rule.

Implementation already exists here: https://github.com/thg-consulting/inspectortiger/blob/6b9736b22528493cd41d0bf92f9fdcdd6c7cd129/inspectortiger/plugins/upgradeable.py#L165 But, it should be adapted to our API.

## Reasoning

Original idea: https://github.com/thg-consulting/inspectortiger/blob/6b9736b22528493cd41d0bf92f9fdcdd6c7cd129/inspectortiger/plugins/upgradeable.py#L165

**Comment by Amyth07** [comment]
Can I take this?

**Comment by sobolevn** [comment]
Let's start with https://github.com/wemake-services/wemake-python-styleguide/issues/892 and later you can take as many as you with. But, we really encourage one-person-one-issue rule, since it allows to concentrate on the task better.

**Comment by meda123** [comment]
Hi - May I work on this issue? This is my first Hacktoberfest issue :)

**Comment by sobolevn** [comment]
Sure! Feel free to ask any questions you have. And, please, ask for help if you need to 🙂 

Happy Hactoberfest! 🎃

**Comment by meda123** [comment]
@sobolevn Awesome! Thank you. I’m going to read through the Contributing guidelines first so I don’t ask something covered there. Thank you!!

**Comment by sobolevn** [comment]
Hi @meda123! How's it going? Do you need any help?

**Comment by sobolevn** [comment]
On the other hand: we have quite a lot of `map` calls that violates this rule.
But, I actually like it. More than with comprehensions.
So, let's leave it as is.
